src/nvim/quickfix.c

```diff
@@ -4182,7 +4187,30 @@ static void qf_update_buffer(qf_info_T *qi, qfline_T *old_last)
   qf_update_win_titlevar(qi);
 
   qf_fill_buffer(qf_get_curlist(qi), buf, old_last, qf_winid);
-  buf_inc_changedtick(buf);
+
+  linenr_T new_line_count = buf->b_ml.ml_line_count;
```

Comment:
Instead of all the new code (below), could we maybe change `qf_fill_buffer` to use `nvim_buf_set_lines` or `nvim_buf_set_text`, which already call `extmark_splice` + `changed_lines` in the expected way?

I guess not, because it looks like `qf_fill_buffer` is doing extra behavior per-line...